src/components/views/rooms/SendMessageComposer.tsx

```diff
@@ -682,6 +683,38 @@ export class SendMessageComposer extends React.Component<ISendMessageComposerPro
             return true; // to skip internal onPaste handler
         }
 
+        // Safari `Insert from iPhone or iPad`
+        // data.getData("text/html") returns a string like: <img src="blob:https://...">
+        if (data.types.includes("text/html")) {
+            const imgElementStr=data.getData("text/html");
+            const parser = new DOMParser();
+            const imgDoc=parser.parseFromString(imgElementStr,"text/html");
+
+            const imgSrc = imgDoc.querySelector("img").src;
```

Comment:
We should apply some filtering here as a heuristic to guess whether the source is safari, e.g. only handle if the contents is a single image with a blob src, skip otherwise otherwise unexpected behaviour when a user pastes multiple images and only 1 gets sent, or the captions are missing, etc

Reply:
When the contents is multiple images pasted by user, then `data.types` would probably be `["Files"]` rather than `text/html`. I think pasting HTML is very rare and can already act as a filter. What do you think I can do to filter?

Reply:
Not if its `text/html` containing multiple `img` tags? Like when copying from google docs or similar. I think asserting that the HTML is a single image tag and it has a blob src is good enough.